src/components/CompareResults/ResultsMain.tsx

```diff
@@ -91,18 +96,106 @@ function ResultsMain() {
+    useState('Results');
+  const [editComparisonTitleInputVisible, showEditComparisonTitle] =
+    useState(false);
+
+  const handleEditInputToggle = () => {
+    if (editComparisonTitleInputVisible) {
+      setComparisonTitleName(previousComparisonTitle); //revert to previous title
+    } else {
+      setPreviousComparisonTitle(comparisonTitleName); //store current title as previous state
+    }
+
+    showEditComparisonTitle(!editComparisonTitleInputVisible);
+  };
+
+  const slugifyComparisonTitle = (title: string) => {
+    return (
+      title
+        /**
+         *please see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/normalize#examples
+         */
+        .normalize('NFD') // Normalize to decompose diacritics (e.g., é -> e)
+        .replace(/[\u0300-\u036f]/g, '') // Remove accents
+        .toLowerCase()
+        .replace(/[^a-z0-9]+/g, '-') // Convert non-alphanumeric to hyphen
+        .replace(/^-+|-+$/g, '') // Trim hyphens from start and end
+    );
+  };
+
+  const onComparisonTitleChange = (value: string) => {
+    //PR notes: fix for truncated letter at the end of entry
+    const slug = slugifyComparisonTitle(value);
+    rawSearchParams.set('title', slug);
+    setComparisonTitleName(value);
+  };
+
+  const OnComparisonTitleSave = () => {
+    updateRawSearchParams(rawSearchParams);
+    showEditComparisonTitle(!editComparisonTitleInputVisible);
+  };
+
+  const deSlugify = (slug: string | undefined) => {
+    if (!slug) return '';
+    return slug.replace(/-/g, ' ').replace(/^\w/, (char) => char.toUpperCase());
+  };
+
+  const deSlugifiedTitle = deSlugify(comparisonTitleName);
+
+  const buttonIcon = (
+    <img
+      id='edit-title-icon'
+      className='icon icon-edit'
+      src={themeMode === 'light' ? pencil.toString() : pencilDark.toString()}
+      alt='edit-title-icon'
+    />
+  );
+
   return (
     <Container className={styles.container} data-testid='results-main'>
       <header>
-        <Grid container className={styles.titleContainer} component='h2'>
-          <Grid item className={styles.title}>
-            Results
-          </Grid>
-          <Grid item className={styles.subtitle}>
+        <Grid container sx={titleContainerSx}>
+          <>
+            {editComparisonTitleInputVisible ? (
+              <EditTitleInput
+                compact={true}
+                onChange={onComparisonTitleChange}
+                onSave={OnComparisonTitleSave}
+                handleToggle={handleEditInputToggle}
+                value={deSlugifiedTitle}
+              />
+            ) : (
+              <>
+                <Grid component='h2' item className={styles.title}>
+                  {deSlugifiedTitle}
+                </Grid>
+                <Button
+                  name='edit-title'
+                  aria-label='edit title'
+                  startIcon={buttonIcon}
+                  className='edit-title-btn'
```

Comment:
same as above, you don't need name/aria-label/className here

Reply:
Looks like I've been mixing aria-lable with aria-description :/ I'll make note of this for the future.

Reply:
yeah, `aria-label` is a way to set an "accessible name", but it's already set by computation here thanks to the button text :-) the devtools accessibility panel can be very helpful to look at that computation.

Remember the first rule of aria: do not use aria if you can avoid it :-)